### PR TITLE
Create SectionExitHandler interface

### PR DESCRIPTION
**Pull request by takejohn**
### Description

* Add an interface `SectionExitHandler`, having one method `exit(Event)`.
* Change `LoopSection` to implement this.
* Change the 'exit' and 'return' effects to check if the enclosing nodes are SectionExitHandlers instead of checking for LoopSections.

This PR enables addons to add sections which don't need to hold loop counters doing something when exited.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

**Review by sovdeeth — APPROVED** [review]
Looks like a nice change to me!

**Comment by APickledWalrus** [comment]
I'm not sure about the name. What do you think about something like `ExitableSection`

File: src/main/java/ch/njol/skript/lang/ExitHandlingSection.java
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/lang/ExitHandlingSection.java:1]: I agree with Pickle, I think the name `SectionExitHandler` would be better

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
Nice change

File: src/main/java/ch/njol/skript/lang/ExitHandlingSection.java
Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/lang/ExitHandlingSection.java:1]: yeah something like that is probably a better name giving that it's not actually a section itself

**Review by AyhamAl-Ali — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]

**Comment by APickledWalrus** [comment]
Thanks for this!

**Pull request merged**